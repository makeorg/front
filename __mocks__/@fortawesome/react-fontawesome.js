
import React from 'react';

export const FontAwesomeIcon = ({ icon }) => (
  <svg
    lib="@fortawesome/react-fontawesome"
    icon={icon.iconName}
    prefix={icon.prefix}
  />
);
