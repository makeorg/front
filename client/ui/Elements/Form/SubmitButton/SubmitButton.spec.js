import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { faThumbsUp } from '@fortawesome/free-regular-svg-icons';
import { SubmitButton } from './index';

describe('SubmitButton', () => {
  const defaultProps = {
    formName: 'foo',
    label: 'bar',
    icon: faThumbsUp
  };

  it('must match the snapshot with default Props', () => {
    const component = renderer.create(
      <SubmitButton {...defaultProps} />
    );
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer.create(
      <SubmitButton {...defaultProps} tabIndex="-1" />
    );
    const PositiveTabIndex = renderer.create(
      <SubmitButton {...defaultProps} tabIndex="0" />
    );
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });
});
