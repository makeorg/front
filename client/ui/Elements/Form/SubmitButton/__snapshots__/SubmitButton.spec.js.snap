// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton must match the snapshot with default Props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-size: 14px;
  line-height: 18px;
  border: none;
  border-radius: 30px;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 10px 15px;
  color: rgb(255,255,255);
  background-color: rgb(232,24,68);
}

.c0 svg {
  fill: rgb(255,255,255);
}

.c0:disabled {
  background-color: rgb(118,118,118);
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-right: 5px;
}

@media (min-width:767px) {
  .c0 {
    font-size: 16px;
  }
}

<button
  className="c0"
  disabled={false}
  form="foo"
  type="submit"
>
  <span
    aria-hidden={true}
    className="c1"
  >
    SvgThumbsUp
  </span>
  bar
</button>
`;

exports[`SubmitButton must return the diff between when disabled 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <button
    className="hs2zb-6 kFpyWt"
-   disabled={false}
+   disabled={true}
    form="foo"
    type="submit"
  >
    <span
      aria-hidden={true}
`;
